### ISSUE TITLE
Thumbnail support

### ISSUE THREAD

**Issue by bperel** [te-online/files_linkeditor#93]
Thank you for this app. Would there be a way to specify a custom thumbnail image so that when viewing URL files in the gallery it's easier to see which URL is which?
In my use case, I store links to YouTube videos on my Nextcloud instance and it would be nice to be able to specify each video's YouTube thumbnail.

**Comment by te-online** [comment]
Hey 👋 There are currently no plans to implement this, as I first have to find the time to port the app to the new Files API for NC 28 😓 I think it's a nice idea, but have no idea how we'd implement this 🤔 We'd need to reach out to each service to fetch the favorite icon and then store it somewhere somehow? Do you have any ideas?

**Comment by bperel** [comment]
> We'd need to reach out to each service to fetch the favorite icon and then store it somewhere somehow? Do you have any ideas?
I don't mean to populate the thumbnail automatically: only that the user would be able to specify the URL of the thumbnail manually.
In order to populate the thumbnail automatically it would be indeed a lot of trial and error so that's why I didn't suggest it.

**Comment by te-online** [comment]
Hm... Okay :-)

Are you thinking about saving a public URL? 🤔 And you'd want to manually apply the URL to all your link files? 😅

**Comment by bperel** [comment]
> Hm... Okay :-)
> 
> Are you thinking about saving a public URL? 🤔 And you'd want to manually apply the URL to all your link files? 😅

Yes, in the same way that some file explorers allow you to set metadata to individual files

**Comment by te-online** [comment]
Sounds great 😊 But I'm not really sure how to do this in Nextlcoud 🤔 

Maybe if you could get in contact with someone from the Nextcloud "Files" app team (probably the core team), they could help you with that 🙂 

I could probably store the metadata in the file itself, but then we'd need to load each single file to display the thumbnail in the files list.

**Comment by bperel** [comment]
I understand. You can close this issue if it sounds cumbersome :-)

**Comment by te-online** [comment]
It does, but that doesn't mean it's a bad idea... 😄
Thanks for your understanding. I will leave it open for a while, maybe someone will turn up to implement it 😉
